File: solutions/go/raindrops/1/raindrops.go
package raindrops

import (
	"fmt"
	"strings"
)

// Convert returns the corresponding sound of n
func Convert(n int) string {
	var sb strings.Builder
	if n%3 == 0 {
		sb.WriteString("Pling")
	}
	if n%5 == 0 {
		sb.WriteString("Plang")
	}
	if n%7 == 0 {
		sb.WriteString("Plong")
	}
	sound := sb.String()
	if sound == "" {
		return fmt.Sprint(n)
	}
	return sound
}

File: solutions/fsharp/raindrops/1/Raindrops.fs
module Raindrops

let convert (number) =
    let mutable sound = ""

    if number % 3 = 0 then
        sound <- sound + "Pling"

    if number % 5 = 0 then
        sound <- sound + "Plang"

    if number % 7 = 0 then
        sound <- sound + "Plong"

    if sound = "" then
        sound <- sprintf "%d" number

    sound
